File: THCS/2022_2023/DakNong_2223/Bai2/PASSWORD.py
def is_prime(n):
    """Kiểm tra số nguyên tố."""
    if n < 2:
        return False
    for i in range(2, int(n**0.5) + 1):
        if n % i == 0:
            return False
    return True

def sum_of_digits(n):
    """Tính tổng các chữ số của một số."""
    return sum(int(digit) for digit in str(n))

def find_special_primes(l, r):
    """Tìm các số nguyên tố đặc biệt trong khoảng [l, r]."""
    special_primes = []
    for i in range(l, r + 1):
        if is_prime(i) and is_prime(sum_of_digits(i)):
            special_primes.append(i)
    return special_primes

# Đọc dữ liệu từ file PASSWORD.INP
with open("PASSWORD.INP", "r") as input_file:
    l, r = map(int, input_file.readline().strip().split())

# Tìm các số nguyên tố đặc biệt
special_primes = find_special_primes(l, r)

# Ghi kết quả ra file PASSWORD.OUT
with open("PASSWORD.OUT", "w") as output_file:
    output_file.write(" ".join(map(str, special_primes)) + "\n")

File: THCS/2022_2023/DakNong_2223/Bai2/PASSWORD.cpp
#include <iostream>
#include <fstream>
#include <vector>

using namespace std;

// Hàm kiểm tra số nguyên tố
bool isPrime(int n) {
    if (n < 2) return false;
    for (int i = 2; i * i <= n; i++) {
        if (n % i == 0) return false;
    }
    return true;
}

// Hàm tính tổng các chữ số của một số
int sumOfDigits(int n) {
    int sum = 0;
    while (n > 0) {
        sum += n % 10;
        n /= 10;
    }
    return sum;
}

int main() {
    // Đọc dữ liệu từ file PASSWORD.INP
    ifstream input("PASSWORD.INP");
    int l, r;
    input >> l >> r;
    input.close();

    vector<int> specialPrimes;

    // Tìm các số nguyên tố trong khoảng [l, r]
    for (int i = l; i <= r; i++) {
        if (isPrime(i) && isPrime(sumOfDigits(i))) {
            specialPrimes.push_back(i);
        }
    }

    // Ghi kết quả ra file PASSWORD.OUT
    ofstream output("PASSWORD.OUT");
    for (size_t i = 0; i < specialPrimes.size(); i++) {
        output << specialPrimes[i];
        if (i < specialPrimes.size() - 1) {
            output << " ";
        }
    }
    output << endl;
    output.close();

    return 0;
}
